form/form_test.go

```diff
@@ -295,44 +289,6 @@ func TestAppendTo(t *testing.T) {
 	}
 }
 
-func TestAppendTo_DuplicatedNames(t *testing.T) {
-	arrayVal := [3]string{"1", "2", "3"}
-	sliceVal := []string{"1", "2", "3"}
-
-	testCases := []struct {
-		field string
-		data  *testStruct
-		want  interface{}
-	}{
-		{"array[]", &testStruct{Array: arrayVal}, sliceVal},
-		{"array_ptr[]", &testStruct{ArrayPtr: &arrayVal}, sliceVal},
-		{"slice[]", &testStruct{Slice: sliceVal}, sliceVal},
-		{"slice_ptr[]", &testStruct{SlicePtr: &sliceVal}, sliceVal},
-
-		// Tests slice nested inside of map nested inside of another map
-		{
-			"map[foo][bar][]",
-			&testStruct{Map: map[string]interface{}{
-				"foo": map[string]interface{}{"bar": sliceVal},
-			}},
-			sliceVal,
-		},
-	}
-	for _, tc := range testCases {
-		t.Run(tc.field, func(t *testing.T) {
-			form := &Values{}
-			AppendTo(form, tc.data)
-			values := form.ToValues()
-			//fmt.Printf("values = %+v", values)
-
-			// This is the only difference between this test case and the one
-			// above: we used square brackets to grab a []string instead of
-			// just a single value.
-			assert.Equal(t, tc.want, values[tc.field])
-		})
-	}
-}
```

Comment:
I removed this test because it's largely here to make sure that duplicate keys are tolerated — duplicate keys like you'd see for an array encoded without integer indexes. Now that we do have integer indexes everywhere, this functionality is no longer important.